scripts/database.js

```diff
@@ -132,33 +133,87 @@ export class Database {
         );
     }
 
-    static async create() {
-        const db = await openDB('MPW', 1, {
+    /**
+     * Migrates from local storage
+     * @param {IDBPDatabase} db
+     */
+    async #migrateLocalStorage() {
+        if (localStorage.length === 0) return;
+        const settings = new Settings({
+            analytics: localStorage.analytics,
+            explorer: localStorage.explorer,
+            node: localStorage.node,
+            translation: localStorage.translation,
+            displayCurrency: localStorage.displayCurrency,
+        });
+        await this.setSettings(settings);
+
+        if (localStorage.masternode) {
+            try {
+                const masternode = JSON.parse(localStorage.masternode);
+                await this.addMasternode(masternode);
+            } catch (e) {
+                console.error(e);
+                createAlert(
+                    'warning',
+                    'Failed to recover your masternode. Please reimport it.'
+                );
+            }
+        }
+
+        if (localStorage.encwif || localStorage.publicKey) {
+            try {
+                const localProposals = JSON.parse(
+                    localStorage.localProposals || '[]'
+                );
+                await this.addAccount({
+                    publicKey: localStorage.publicKey,
+                    encWif: localStorage.encwif,
+                    localProposals,
+                });
+            } catch (e) {
+                console.error(e);
+                createAlert(
+                    'warning',
+                    'Failed to recover your account. Please reimport it.'
```

Comment:
If, in whatever case the account migration fails, I think it'd be better to popup the "Export Private Key" modal alongside an error like "Failed to migrate data, please backup and reimport your wallet", in the very odd case that a user DOESN'T already have a backup.

Otherwise the *'lesser brained'* users so to speak could potentially lose funds from this failure.